
If you don't know anything about coding then this might be nothing more than copying and pasting code.

So I will quickly explain:


```python
a = 10
print("a is a number equal to:", a)

a = "Nguyen"
print("a is now a string which returns:", a)

a = [1,2,3,"cat","5","meow",7]
print("a is now list which returns:", a)
```

    a is a number equal to: 10
    a is now a string which returns: Nguyen
    a is now list which returns: [1, 2, 3, 'cat', '5', 'meow', 7]


For loops can execute a task a set number of times.
They can also be used to pull out every item in a list (or other structure) one at a time. 
In this example, I use the list we defined in the previous section. The loop starts at the first variable in the list, then moves on to the next piece of indented code. The if statement checks whether or not that value is an integer, then prints something depending on the result. After that, the code block is repeated using the second value in list a.

Notice that i gets assigned to the value of the object in the list every loop.


```python
b = 0
for i in a:
    if type(i) is int:
        b += i
        print(i, "is an integer")
    else:
        print(i, "is not an integer")
print("The sum of all integers in the list is", b)  
```

    1 is an integer
    2 is an integer
    3 is an integer
    cat is not an integer
    5 is not an integer
    meow is not an integer
    7 is an integer
    The sum of all integers in the list is 13


Integers, strings, and lists are basic types of objects that you will always use, but there can be any type of objects. Many python modules will have their own types of objects with unique data structures. 


Objects are acted on by methods, and they will output another object (usually). You can assign those objects to variables and send them into other methods.

# Biopython

Biopython is a massive python module designed for bioinformatics!

   **Note on Biopython in the cluster and version control**
The cluster has a few random versions of python to choose from, and the packages within them are maintained by someone else. As a result, only one of them has BioPython installed: python/2.7.8

You can develop scripts for this version of python, but biopython is very out of date and you can't use the 'qblast()' function, nor 'Entrez.read()', and probably some others that I haven't tried. The best way around this is to install your own updated version of python in a virtual environment, which lets you add your own packages!

After logging into the cluster:

    module unload python
    module load python/3.6.3
    virtualenv py-3.6.3
    module unload python

This creates a directory with a new python in it. Activate by running the activate file 

    source /py-3.6.3/bin/activate
    
When you're in the virtual environment, you can install new packages using `pip`

    pip install biopython
    
    
To exit the virtual environment type

    deactivate
    
    
Our lab has started a group virtual environment so we can all have the same package base. You can activate it with

    source /UCHC/LABS/Simon/venv/py-3.6.3/bin/activate
    
   or if that part of the cluster is down again use mine:
   
    source /home/CAM/jvailionis/venv/py-3.6.3/bin/activate
   
   (or make your own, it's easy)

### Reading fasta files in python using SeqIO
   The SeqIO module is designed for handling fasta files. There are two methods to read fasta files with this module


```python
from Bio import SeqIO
```

   SeqIO.parse() generates an object with two important components: 'seq' and 'id'


```python
for sequence in SeqIO.parse("/UCHC/LABS/Simon/Python_Test/mystery.fasta", "fasta"):
    print("Name:", sequence.id)
    print("Sequence:", sequence.seq)
```

    Name: Uninformative_Sequence_Name_1
    Sequence: AAACACACCTACCGTACAACAGCCGATACAGACGATACTACACAACAGCTTAAATTTGCGGATAATGAACAAGAGAACCCAATACAGTTCTTGAAGCATATGGAAATGAATATGGAAGCAATTTTATGATGCGATAAACGATAGTGAAAAAATAAATTGGATTAGCTGTCATCATAAGGGACCGCATCAGATTGGTTTTCGGTCATTCAAGATAAAGTGTAACCTTAAATCCAAAATGATGAAGAGAAGAGAAATGAAACTTTCATTGTCGAAAAAGACAAATTATAATAAATGTTCTTCCAATAATAACATGAATTCCGTGGAATCCTGTTGCCATGAAAAATGTTGATCCATAAATTGAATCAGAAATACAAAATGGGGATTCATAATATTCATATCCTTGCAAAATAGTGAAATATACACCTAACAATAAAGTCAATATAATTCTATAAATAGATTCATCATATTTTTTTTCTATAATCGAATGGTGAGCTCAAGTAATAGTTACACCAGAACATAATAAAATTATAGTATTTAATAATGGAACATGTATTGGATTAAATGGCTTAATACCGGCAGGAGGTCAAATTATTCCAATTTCTATAACGGGTCTTAATCTTCTATGAAAAAATCCCCAAAAAAAAGAAATAAAAAAAAAAATTTCTGAAATAATAAATAATACTATTCCAATCCTTAAACCCCTTACAACCGTGACTGTGTGTAATCCTAAAAACGTTCCTTCTCGTGAGATATCACGTCATCATTGAATTATTGTTAAAATTAAAATTAAAAATCCAATAATTAATAAAATAAAACTATATATATGAAATATTTTTACTATTCCAGATACAAAAGTAAGTGCTCCAATAGACCCAGTTAAAGGTCAAGGTCTATAATCAACTAAATGAAAAGGGTGGTTATTCATAAGAAATTTCTCTATAATATAAAGTGCTAAGAACTGTAAAAACATATGATTGAATTAAAGAAACAGCCGATTCAAACATTATTAATATAATTTGAATGAAAATAATTAAATATGTGCATATTCTTATATTAATTGAACTATTTCCTAATAATGTTATCAATAAATGACCTGCAATTATATTAGCTGTTAAGCGAACAGCTAATGATCCAGGTCGAATTATATTTCTAATTGTTTCAATACATACTATAAATGGTATTAAAACAGTAGGAGTTCCCATTGGAACTAAATGACAAAATATATGATTAGCATTATTTAAAAATCCAAATAAAATAAAAGTCATTCATAATGGCAAAGCCAATCTAATTGTAAATGTTAAATGACTTGATCTAGTAAAAATATATGGAAACAATCCAAGAAAATTATTTAATATAATAAATATAAATAAAGAACAAAATAATAAAGATCTTCCTAAAGATCTTTTTCCCATTAATATATTCAATTCAAGTCTTAAAGTGTTTAAAATTTTGATTAAAGTTATTAATCTTCGATTAGGAACCACTCAATATACTAAAGGAATAAAATATATTAAAGATGAAGATCTAATTCAATTAAATGAAAACATACCTGTTGATGGATCAAATGTCGAAAATAAATTTCCTATCATATTCAGTTAATTGTTTTAATTAAATTATTAGATAATTTAATCTTAGGTATAACCAAAAATGAAAAATAAATTACAGTCATAATTAATAAAATAATTATAATAAAATAAAAAAATAAAATTACTCAATTTATTGGAGATATTTGAGGAATGGAAAAATACTATTTAAGTTTCATTATTCTGACAGAATAATATTTTATATAAAATAATTTTTCCATTAAGAGTAATCGTTAATTACTATAAATTGGTTTAAGAGACCATTACTGACTTTCAGTCACTTAATGAACAATTAATTAACCAATTAATAAAATATTTTGAGCTAACTCTTTCAATAACAATGGGTATAAATCTATGATTTGCGCCACAGATTTCTGAACATTGCCCATAAGATAATCCTGGTCGACTTATAAACATTCTGGCTTGATTTAATCGACCAGGGACTGCATCCACTTTCACTCTCATAGAAGGCATAGCTCATGAATGAATAACATCAAAAGATGAAATCAATAAACGAATTTGAGTATTGAATGGTAAAACTGTTCGATTATCCACATCTAATAAACGAAATTCACCTATCAATAATTCATTATATGATTTTATATATGAATCAAATTCAATTATCATAAAATCCGAATATTCATAAGATCAATATCATTGATGACCAATAATTTTTAAAGTTATTAAAGGCTTTCTAACTTCATCGAGTAAATAAAGTAATCGTAAGGATGGAAATGCAATAAAGACTAAAGTCAAAGCAGGAAGAAAAGTTCAAAAAAATTCAATTATTTGTCCTTCTAATAATAATCGATTAACAGTAGTATTAAAAATTAATATTACTATCATATAAGCAACAATGGAAGTAATAACAATCAAGATAATTAAGGTATGATCATGAAAAAATATTAATTGTTCTATTAATGGAGATATAGCATCTTGTATATTAATATATGACCAATTGCAAATTTCTAATAAAAAAATTATTTTATATATGAATCTTAAATTCATTGCACTTATTCTGCCATATTAGAATTGAATTAATATGGGAATTTCTCTATATGAATGTTCTGAGGGGGGATATTTTTGTAATCATTCAATTGATGATATTATATTATTTGAATATAAAATTTTTCGTTGTGAAATAAATCTTTCCCACACAATAAATAACAATAATAAAATTCCAATTAATGAAATTCTTCTTCCAATAGAAGATAACACATTTCATGATATATAAGCATCTGGATAATCTGAATATCGACGAGGTATTCCACTAAGTCCCAAAAAATGTTGAGGAAAAAAAGTTATATTCACACCAATAAATATAACAATAAAATGAACCCTTAATCAATTAGAATTTAATGATATCCCAGTAAATAATGGATATCAATGAATAAATCTACCTAAAATTGCAAACACTGCACCTATGGATAAAACATAATGAAAATGTGCTACCACATAATATGTATCATGTAAAACAATATCAATAGATGAATTAGCTAAAATTACACCTGTTAATCCTCCAATAGTAAATAAAAACACAAATCCTAATGCTCATAAAATAGAAGGTGAGATCTTAATTTTAGTTCCTCTAAGAGTTGCTAATCAACTAAAAACTTTAATTCCAGTAGGAACAGCAATAATTATTGTTGCTGATGTAAAGTATGCACGAGTATCAACGTCCATTCCAACGGTAAATATATGATGAGCCCAGACAACAAAACCTAAAATTCCAATAGATATTATTGCATAAATTATTCCCAATGACCCAAATGACTCAACCTTTCCTCTTTCTTGAGTAATAATATGTGAAATTAACCCAAATCCTGGTAAAATCAAAATATAAACTTCAGGATGACCAAAGAATCAAAATAAATGCTGATATAAAATGGGATCCCCCCCTCCTGAAGGATCAAAAAAGGAAGTGTTTAAATTACGATCAGTCAATAATATTGTAATAGCTCCTGCTAAAACAGGTAGGGATAATAATAACAAAAAAGCAGTGATTAAAACTGCTCAAACAAATAATGGTATACGATCCATGAAAATTCCGACTGAACGTATATTAAAAATTGTTCTAATAAAATTTACCGCACCTAAGATCGATCTTGCACCTGCAAGATGAAGAGAAAAAATAGTTAAATCAACACAAGAACCCGAGTGCGCAACACTACTCGATAAAGGTGGGTATACCGTTCATCCAGTACCAGCACCTCTATCAACTATTCTCCCAATTAATAAAAGTGTCAAAGAAGGTGGTAGTAATCAAAAACTTATATTATTTATACGTGGAAAAGCTATATCTGGAGCCCCAATTATTAAAGGAACTAATCAATTTCCAAATCCACCAATTATAATAGGTATAACTATAAAGAAAATTATAATAAAAGCATGGGCAGTGACAATAACATTATAAATTTGATCATCTCCAATAAATGATCCAGGAATTCCTAACTCAACACGAATTAAAACTCTCAAAGTTGTACCAATTATTCCTGACCAAATTCCAAAAATAAAATATAATGTACCAATATCTTTATGATTAGTTGAAAAAAACCATTTATTCATAATCATAATTTTATTGGGATGGCTGATAAAAGCAATAAATTGTAAATTTATTAATGAAAATAATATTCTCCCAATAGTGAGTCTTATATTTAATTATAATATTAAATTGCAAATTTAAAGGTGATTTAAAGTAAATCTAAGGCTTATAATAATTATATTTTTAACTTTGAAGGATAATAGTTTCTTTAACTTAAAACCTTAATTACCAACAGTAAGTAAAGAAATTCTCATCAATCCAATTAATAAAAATCTACCCAAATAAATTAATCTATTCATCTTAATAAATAATATCTTTATTGTAACTATATTTATTATTAACCCAGTTATTACAATGCGTAAATAATAAAACAAAACCAACAATGCTGTTATTAATATCATTAAACACAATATGTATATATTATTAATTAAAAGACAATAAATTATTATAAATTTAGGAAAAAAACCAAAGAATGGTGGTAACCCACCTATTGATAAGAAAACAATTATTACCAAAAAAGTCTTAATTTTGTTCCCAAATATTAAGAAGAATTGATTTAAATAGTTTAAATTATTTATATCCAATAAATAACAAATTAAATATAAAGAAAATGAATAAATTAAAAAGTAAATTAATCAAGAATAATTCATTATCATAACTCCCACAAAAATTCATCTTAAATTATAAATTGAAGAGTATGCTAAAATTTTTCGAAACGAAGAGTAACATAGACCACCTAATGAACCCCATAAACAAGAAAAAATAATAAATATAATTAATCTTATATCAATATAACTCAATAAAACTAAAGGAATAATTTTTTGAATTGTCATAATTATAAAAGAAACTATCCAAGATAGCCCATCAATTACTGAAATATATCAATAATGAAATGGAGGACAACCAATTTTAATTATTAATCTAATTATAATAAAATAGTTAAATTCTACGCATACATTTATAATAACCATTGAAAGAAATAGTAGTCTTGAACCTATTCTTTGTACAATAAAATACTTAATCATAGATTCGGAAGAATAAACTCTTATTTTGTTTAATATAAAAGGCAAAAAAGAAATTATGTTTACCTCAATTCCAATCCAACAACCCAGCCAATTGTTAGAAGAAACGGATAACATAATCCCTAATAATAAGAACGCCATAAATATTATATAAGAGGAATTTTTTTTAATTAAAAAGAGGATTTTATTTCCATAAATGAGGTATGAACCCATTAGCTTAAGTTTAGCTTATCTTTTATGTTGTAGTTTATGAAGAACATTAATTTTTGATATTAAAAGACAGTTTAATTCTGACAACATATAATAAGAGTATTATATACATAATTTATTCTATCAAAATAACCCTTTTATCAGGCATCTTATTATCTATAGCCACTACCATAATGTGTATTTTATTGATAAAATAATAGCATTAGGTGTAACGATATATCACACTGGCATAACTCATTATACTCTATCTACACGAACTTCCCTATCTATAGCCACTACCATAATGTGTATTTTATTGATAAAATAATAGCATTAGGTGTAACGATATATCACACTGGCA
    Name: Uninformative_Sequence_Name_2
    Sequence: AATTTTATTTTTTAGTAAAGATTTACTCATATTAAGATATAATTTGTGATTGGATAATATCTCATACGGATTAATTATTCTTACTGTATGAATTTCATTTTTAATAATTAATTCAAGACCAAATTATAAACATAATAATATGGATGTATTTTTATTCATAGTAATATTATTAATAATTTTATTAATTTTATCATTTTGTTCTTATAACTTAATATTATTTTATATTTTCTTTGAATCAAGATTAATTCCCACAATAATTATTATTATGGGATGGGGCTATCAACCTGAACGAGTTAATGCTAGATATTATTTATTATTTTATACTTTATTTGCATCATTTCCATTTTTAATTTCCATTATTTCATTATATATAAATAATTTAACAAATATTATGTTATTAATATATTCAAGAAATAAATTTATTTATTTAAGAATAATTTTAGCATTCATAGTGAAAGTTCCACTATTTATATTTCATTTTTGATTACCTAAAGCTCATGTTGAAGCTCCGGTTTCTGGTTCAATAATTCTAGCTGGAATTCTATTAAAATTAGGCACATATGGATTAATTCGTTTAATGTTTATTATGCCGATAATATTTAATAAATATAGATTTATTTGAATTTCAATTCCTATAATTGGAGGAATTATAATTAGATTATTATGTATAATACAAATTGATATTAAATCAATAATTGCTTATTCATCCGTTGCTCATATAGGTCTGGTTGTTGGGGGGATTATAACAATAAATACTTGAGGGTTACTAGGTTCATACTATATAATTATTGGGCATGGGCTATGTTCCTCAGCTTTGTTTTGCTTAGCTAATATTAGTTATGAACGAATGGGATCTCGCAGAATATTAATTAATAAAGGAATGTTAAGATTTATACCATCAATAACTTTAATATGATTTTTATTATCAGCAAGAAATATTTCATGTCCGCCAACAATTAGATTAGCTGGAGAGATAATAATTTTAAATAGTTTAGTGTCTTGAAATTCATTAAGCATAATATTTTTGTCTATTTCCTCTTTTCTTTCAGCTGGCTATAGATTATTTTTATATTCATATACTCAGCATGGAACATTTTTTTCAGGCTCTTATTCATTTAATTATGGAGTTGTTCGGGAGTTTTTTTTAATCATAATACATTGAATTCCGTTAAATTTTATCATTTTAAAGATAACATTATTAATTTATTCAATTAGTTTAACATAAAACATTAAATTGTGGATTTAAAAATATAAATTTATATTGGATCATGAATAATAAAATCAATATATACTTCTTCTTGATCTTGATAGTTTTAAGATTAATAATAATAATTTTATCTATAAATATAATTATTTTCGATTATGTTGTAATAATAGAATGAATAATTGTTAGAATTAATTCATGTAATATTTATATAACTATAATTTTTGATTTTATATCAACTATATTTTTATCAACAGTGATATTTATTTCATCAACAGTGATGTTTTATAGAGGTACATATATAGAGAATGATAAAAATATTAGACGATTTATTTATATTATTATAATGTTTGTAATATCAATAATTATATTAATTATTAGTCCTAATATAATTAGTATTATTGTTGGATGAGATGGACTTGGTCTTGTCTCATATTGTTTAGTAATTTATTATCAAAATTTAAATTCAGCTAATGCAGGAATATTAACCGCTTTAATAAATCGAATTGGAGACGTAATGATTTTAATATTAATTGCATGAATATTAAATTTTGGTTCTTGAAATTTTTTATCATTTATTAGTAAAATAGAAATAAATTTTATATTAATTTTTATTATTATTGCAGGGTTTACCAAAAGAGCACAGATCCCATTCTCTTCTTGACTTCCAGCAGCAATAGCTGCTCCAACACCAGTATCTGCATTAGTTCACTCATCAACATTAGTTACTGCTGGGGTATACCTAATAATCCGATTTAGAAAATTAATTCTTACATTAAATTTTATTCAGTTTTTCTTAATATTATCTATCATAACTATAATTATGTCTGGTATTGGGGCAATTTTTGAATTCGATTTAAAAAAAATTATTGCATTATCAACATTAAGACAGCTTGGGATTATGATATCAATTTTAATATTTGGTTATCCAATATTGTCATTTTTTCATCTTATTATTCATGCTTTATTTAAAGCATCTTTATTTTTATGTGCAGGTGTATTAATTCATAGATTAAATAATAATCAAGACATTCGACTTATAGGATGTATAAGTTATAATATACCCTTAACAATAATGTTAATAAATACTGCTAATTTATCATTATGTGGAATTCCATTCATAAGAGGATTTTACTCAAAAGATTTAATCATAGAAATAATATGCTCAAATCATATCAATATATGAATTATTATATTAATATATTTAGGAATTGGCTTAACATCATTTTACTCTGCTCGATTAACTTATTTTTCTATAAATATAAATTTTAATTACTATAGTTTTAATTCATTAAATGAACCATTTAATAATATACTAAAAAGTATTTTAATACTATCTATTTATTCAATTATTTCAGGATCTATAATTACATGATTAATATTTAATAATCCAATTATTATCTCTCTTCCAATAGAAGGGAAAATAATAGCTTTATTTATATCTATAATCGGATTATTAATTGGATATGAAATTTCATCATTTTCAAAAACAAAAAAGAGAATAATGAAGGAATTTCTTGGAACTATATGATTTATAAAACAAATTTCAACGTTCCATAATCAATTTTTTTTATTAAAGAATTCTTATATATATCAAAAATCAATTGATTTAGGTTGAGGAGAAAAACTAGGGGTCCAAGGATTAGTTCAATTAATTAAATTTATGATATTTATTAATATTAAAATAAGCAAGAATAATTTTAAAATTCAAATAATAGCATTTATTTTATGGATCTTAATTATATTATATTTAAATAGCTTAACAAGAGCATAATATTGAAGATATTAAGGTGATAATTATATTTTTAAATATTTATAAGACAATGCCTCTCTTTCCAATGAAAATGAAATGTTTTAAAAACTTTATAAATAAGGAAAAAACGGATTTTAACCGCTTTAAAAACCAGTTAGCAGCTGCTTTTAACCTCGTTTCCTTTTAATTGGAAATACATTCAATTACCTTATTAACAATAAAGTGCCTCTATTTTGGCTTCAATTAAAACAATGGAATATATATTCTTTTTTTAGGTCGAAACTAAATGTATTAATTACTTCATTATTACAAGAATGGTTGATAACAACACCTATTGATTGCAATTCAATTATTATAATTAAATACATCCCTCATTAAACTCAATTTAATATATTATTTATTCATTCATGATATAATCCTAAAATCAATAATAATAAAAATACCATTATAATTATATAAATTTCAATTATATTAATTGAAGATATAATGACAACAACAGGTAAAATAATAACTAATTCAACGTCAAAAATCAAAAAAATAACCGCAATTATAAAAAAATGTCTAGAAAATGGAATCCGTGAGGATTCAAACGGATCAAATCCACACTCAAATGGTGATGATTTTTCACGATCAATGATTGACTTAATAGAAATAATATATAAAATAATTATTAAGATAGATAATACAAATAATAAAATAGCAGAATATATTATAAT
    Name: Uninformative_Sequence_Name_3
    Sequence: GCTACAATTTTACTGGGTTATTGGTTTGTGGGTATAGTTAATGGGGCTAATATTTGATGGGGCTAGTAGTACCCAGTTGGTCGTAATTGGGTATATTAATTGATGGTCCATTAATCATTGTATCATTTTAATGGAGATGTTCTTTAGTAGTTTGGGGCAATTATTTTATTGTGTTGTTAGTTAATAGGGCTACTAATTAATGGGCCTCCTAGGTAATGGTACTATTGTGTTATTGCGCTATTAGATAATGAGGATAATTAGTTAATTAGGCCATTAGTTAATGGGGTACTAGTTTATTGGACTATTATGTAATGGAGAGGGTTAGTTTACTGTGCCACTAATTAATTGTAGTTTATTGGGCTATTAGTCATAGGCCTACTTGCTAATGTGTCTGTTAATTTATTGGCTATCAGTTAAATTAGGAATATGCTTAATTATTCAAATTTTAACGGGTCTCTTCTTGTCAATACACTATAATGCGGATGTAATAAATGCATTTAATAGATTAAGACATATTTGTCGTGATGTTAATTACGGTTGAACACTACGTATTTCTCATGCTAATGGAGCTTCATTATTTTTTATTTGCATTTATTTACATGTAGGACGAGGTTTATATTATGGTTCTTATAAGTATCTCGAAACTTGATCAGTAGGAGTAATTATTTTATTGATCTTAATAGCCACAGCTTTCTTAGGGTATGTCCTTCCCTGAGGACAAATATCATTCTGAGGGGCAACGGTAATTACAAATTTATTATCGGCTATTCCATATCTTGGAACAATTTTGGTGAATTGGATTTGAGGGGGGTTTGCTGTTGACAATGCCACTTTAACTCGATTTTATTCTTTCCATTTTATTATGCCATTTATTGTTCTAGCAATAACAATTGTTCATTTAATTTTTTTACATATAACAGGTTCTAATAATCCATTAGGAATTAATTCAAATTTTGATAAAATTCCTTTTCATCCATATTTTTCTACTAAGGATATATTAGGATTTATATTAATATTTATTTTTTTCTTTATTTATATTATAATTGAGCCTTATAAATTAGGGGATCCGGATAACTTTGTCCCGGCTAATCCATTAGTAACACCAAAACATATTCAACCAGAATGATACTTCTTATTTGCATATGCAATCTTACGATCTATTCCTAATAAATTAGGAGGTGTTATTGCTTTATTTATGTCAATTTTTATTTTAATTTTTGTACCCATAGTAAACAACTCAAAATTTATAGGATTAACAAATTATCCAATTAACCAAATTTTATTTTGATATATGTTAATTATTTTAATATTATTGACTTGGATTGGTGCTCGACCAGTAGAATATCCATATATTAAAATTGGAATATTATTAACATTATTATACTTTATATATTTTATTATAGATCCTGTTATTAAATACGTTTGAGATAAATTAATTGATTAAATTAGACAATTAGCTTATCTTAAAGCATTTATTTTGAAAATAAAGAAAAGAATTTTATTCTATTGTCTTAATTAATTTTGACACTAGCAATTTTTAATGAAACTAGATAATAGATTTTATTAATAAAAAAAAAATTATTAAATTTAATGCAACCGGCAAATAAGACTTTCAAGCTAAATATATAAGTTTATCATAACGATATCGTGGCAAAGTACCACGAATTCAAATAAACAAAAATGAAATTAAAGTTAATTGAATAGGAAAAAATAATGAATTAATTTTTCTCCCTAAAAACATGACACAAAATATTATTCTTATAAAAATAATTCTGGCATATTCTGCCAAAAAAATAAAAGCAAATCCCATTCCTCTATACTCTACATTAAATCCTGACACTAATTCAGATTCTCCTTCCGAAAAATCAAATGGAGACCGATTAGTTTCTGCTATAGCAGATGAAATTCAACATATTATTAATGGAAAAAATAAAAATATAAATCATATATTTTTCTGAAAAAAGAAAAAATCAATAATATTAAATCTTTCAATTATTACGACAGGACATAAAATAATTAATACAAGTCTTACTTCATAAGAAATAGTTTGAGCAATTGATCGAATACAGCCTAGCATAGCATAATTAGAATTTGATGCTCAACCAGTAATTATAATTCTATAAACTCTTATTCTAGTACAACAAAAAAAAAATAATATTCCTATATTAAATGAAACACAGTTAATTATGTATGGATAAATAACTCATATTAATAGAGCATTAAATAAACCCAACATTGGACTAATTATATAAATTAAATAATTAGATAATATTGGCATATTTCTTTCTTTTAAAAATAATTTCACAGCATCTGAAAGTGGTTGTAAGATACCAATAAATCCAACTTTATTAGGTCCTTTGCGAGTTTGAATGTATCTCAATACCTTTCGTTCCATTAAAGTCAAAAATGCCACCCCAATTAAAATAAATAAGATCGTAATAAAAATTGTAACTAAATACATATTTCTATATGTAATTGAAATTACATAATTAAATTCTAAATTTAATGCACCTTTTGCTAATATAGAATAATTATATTATTCAATATGGTCCTTTCGTACTAATATTAAAAAATTTTTAGTAGATAGAAACCAACCTGGCTCACGCCGGTTTGAACTCAGATCATGTAAGAAATTAAAGGTCGAACAGACCTAGAATATTTAAAATTTGCTTTAAATTCTTTTCTTAATCCAACATCGAGGTCGCAATCTAATTTATCGATATGAACTCTCCAAATTAATTACGCTGTTATCCCTAAGGTAATTTAATCTTTTAATCAAAATTTTGGATCAATTTAACACTAATTAGTGAATAATAAATTAAAAGTTTTTAATATTTCAACATCACCCCAACGAAAAATACATTAAATTTCTTAATCATCAATTCTAAATTTTAAGGAATTAAAAATATTTTAAACTCTATAGGGTCTTCTCGTCTTACTCAGTAATTTGAGATTTTTTACTCAAAGCTTAAATTCAATAAATTTTAATTAAAAAAGTTAGAATCTCATCCATTCTTTCATACAAGCCTCCAATTAAAAGACTAATTATTATGCTACCTTTGCACGGTCAATTTACCGCGGCCATTCAAATATCATTGGGCAGAATAAATCTTTAATAACTACTAAAAAACATGTTTTTGATAAACAGGTGGAAACTATTTTTGCTTAATTCTTGAATACAATTTTTTAACTTACTAATTAAATCATTTTTATAAAAATAATCCGTTATTTTAATTATCTTAGTATATATCTAAAAAATTTTGAATACATTATAAAATAAAATAAAATTTAACTTATTCAAATAAAAGACTGTTTTTAAGTCTATTTAATTTATATATTAAAACATATTTATAGATATTATATCTAGCTT
    Name: Uninformative_Sequence_Name_4
    Sequence: AGTGTTTGCTATAAAAATATTGGTAATCTCTTATGCTATTCTATTCTTATCTTTATTGTTATATGAAGTATCATTAAACTGAAGAAACTGAATATTAATCATAATAATAATATAGAAATGTTTGTTATTAAGAATATGTATTTTAAACAAAGATATTTTAAATCACTATAATTCGGGAAAAATAAATTCTTAAATATAATTAATTAACCCTGATACAAAAGGTACAAAATTAATTTAACTTTATTATATTGCTCATTTTATCCTTACAGTTTTATAAAAAAATAATTTTTTTAGACAATACTCATACTAGATAATTATTTTCATGAATGCAATATATACATAAAAATCATTATATTACTAATTCAAATTATATTGATATTATCAATAATTCTTGTCAGTGTAAATAACATGCTCACCTTGGCTTTAATTTGCATTCTAGCTACACCTTCCAGTACAACTACTTTGTTACGACTTATCCCATCTTATAATGGGAGTGACGGGCGATGTGTGCATAACTTAGAACCTATATTCAATTAATTATATTTAATATATTTACTTTCAAATCCAATTTCATAACATAATCATAAATGTTTTTCCAAAAATACAATAATGTAATCCATTTTTTCTTCAAAATAAACTGCACCTTGACCTGACATTAAATTAATAAAATTTAAGAAAATTTTCTAATAAAACATTCTTGACAGAGGTATACAAGTATAATTAAAGTTGCATCTATCGTGGATTATCAATTATAAAACAGATTCCTCTAATTAGATAAATTACCGCCAAATTCTTTGAGTTTTAAGATCATATCTATTAATAATCTAGTTATCGTATTTACACTTCAAATAATAGGGTATCTAATCCTAGTTTATATAAAAAGTTTCTCAGACATAATAATAAACACTTATATAAACATTATTTCACTAAAGTTTAAATAATTTTAAATCAAGATTAATTAACTGAAACTAATATAAATTAACTTTAGAAAATCGTATAACCGCGACTGCTGGCACGAGTTTTGTCACCTATAATTATAATTCACTATTTCTAAAATTAATAAATAATAATATTATATACTGTGATATTAATCAATATCTTTTAGATAATCTTTAAATGAAATAAAATTTTACATGTATTATGATTATATATTTAATTCTATAAGCAAGAATAAAACTTTCATATAAAATATGACCATATGTATTTTATTAACCATAAATAGAATACTTTAATCATTATTCACAATATGAAGTATTTTTTTAATAAAAATATTCCTCCCAATTTAATTAATGATTTAATGGCTGGTTAAATCACTACCAATACAAAATATTTTTAATAAAAATATTCCTCCCAATTTAATTAATGATTTAATGGCTGGTTAAATCACTACCAATACAAAGTATTTTTAATAAAAATATTCCTCCCAATTTAATTAATGATTTAATGGCTGGTTAAATCACTACCAATACAAAGTATT
    Name: Uninformative_Sequence_Name_5
    Sequence: TCTGCTGGATTATTCCTTCGTAGGTTCGAATCCTACTCTTCCCACTATGCACTATGCGGGAATAGCTCAGTTGGCAGAGCATCAGCCTTCCAAGTTGAATGTCGCGGGTTCAAATCCCGTTTCCCGCTCAAAAACCAACTTATTTACGCTCATCATGCCGATGTAGCTCAGTTGTAAGAGTACTTTCTTGGTAAGAAAGAGGTCACGGGTTCAACCCCCGTCATTGGCTCTAAAATAAATAAAAAAAAATATGGCAAAAGAAATATTTAAACGTGATAAACCTCATTTAAATATAGGAACAATAGGTCATGTAGATCACGGTAAAACTACTTTAACTGCTTCTATAACAACAGTTTTAGCTAAAAAAGGACTAGCAATAGCTAAAGATTTTAGTTCAATTGATAATGCTCCTGAAGAACAAGAAAGAGGAATAACAATAAATACTTCTCATGTTGAATATCAAACTGAATCACGTCATTATGCTCATGTAGATTGTCCAGGACATGCTGATTATGTTAAAAATATGGTTACAGGTGCTGCTCAAATGGATGGAGCAATTTTAGTAGTAGCTGCAACAGATGGACCCATGCCTCAAACAAGAGAACATATTTTATTAGCTCGTCAAGTTGGAGTACCAAATTTAGTAGTTTTTATGAATAAAGTAGATCAAGTGGATGATCCAGAACTAATAGAATTAGTAGAAATTGAAATTCGAGAATTGTTATCTAAATATGAATATGATGGAGAAAATATTCCAATTATACAAGGTTCAGCATTAGGAGCTTTAAATGGAGAAGAAAAATGGATAAAAAAAGTAGAAGACCTTTTGGAAGCTGTGGATAATTATATAAAAGAACCCATAAGAGATATAGAAAAACCATTTTTAATGCCAATTGAAGATGTTTTTACTATTACAGGACGTGGAACTGTAGCAACAGGACGGATAGAAACATGTATTATAAATACAGGAGATTCTATTGAAATAATAGGAATGGGGGAAAAAAAACTTAATTCTATAGTTACTGGAGTAGAAATGTTTAGAAAAATTTTAGATAAAGGCCAAGCTGGAGATAATGTAGGTTTATTATTAAGAGGGATAGAAAAAAAAGATATCCGTCGTGGAATGGTTATATCATATCCTGGATATATTACTCCTCATAAAAAATTTATGGCTCAAGTATATATTTTAAAAAAAGAAGAAGGAGGCAGGCATACTGCTTTTCATAATAAATATAAACCTCAATTTTATTTAAGAACAACAGATGTGACAGGAACAATTATATTATTAAATAATGTAGAAATGGTAATGCCTGGAGATAACATTACTGTAGAAGTAATACTTTTTCAACCAGTTGCCTTAAGTGAAGGTTTAAGATTTGCAATTCGTGAAGGTGGAAAAACAGTTGGTGCTGGACAAGTAATTAAAATACTTGATTAAAAACGGGTGTAGCTCAGTAGGTTAGAGCATTGGTCTCCAAAACCAAAGGTCGTGAGTTCGATTCTTACCACCCGTGCCTTTATTTAACCTTTATTTAACCAGCCGGATGAAAATTAAAAATAAAATTAAAAGCTATGAAACCGCACATAAAACACGAAATATATACATTTAACATTGATATTATGGTACATAAAACACAACAATATGGGTTCGATTCTCTTTAGCTAACATCTGATGATGAAGTTTAACTTCGAAACGCGTAATTAAATAAATATGATTAACTTCGTAAAAACTTCTTCTGCGGTATTTTTATTGTATATTCACATAAACACAATACGTATTTCGGAAACACTTATAAATTTATTTCCATCTTCAGGTGGTGTGTGTGTTGTAGAGTGAATGTAA
    Name: Uninformative_Sequence_Name_6
    Sequence: TCTGTGGCATCTGCCCATCTGTTGCAGCAACAACTAATATAGCACCATCCATCTGAGATGTCCCTGAGATCATGTTTTTAATATAATCTGCATGCCCAGGGCAATCTGTATGAGCATAATGTCTCTTCTTTGTTTTATATTCTATATGTGTAGCATTAATTGTTATGCCACGTGACTTTTCTTCTGGTGCTCTATCAATCTCACTGTATGGAATATACTTGGTCTCGTTTTCTTTTGACATCAGCTTCGTAATTGCTGCAGTAAGAGTAGTTTTTCCATGATCAACATGACCTATGGTTCCCACATTACAATGGTTTCTACTAATTTTCGTACAGTAACACCTTATCCTATCCGTGCTACGTATAAAACTGTAATTACTAGTATTTAAAACTGTTCTTGAACTTTTCACTATATTTAAATACACGACTGTATGCCTCACACATGACAAAACTATGCTGTTCATTTTCTTCAAATTCACTTCAATCCTTTATTTTTCTTGATTGGTCCCACGCCTATATAAAGACGAGAAAAGGAAGACAGGAATATTAGTGTGCTGTAATGAGACTAAGTATGTTGATATTAAAATGGAAGACGTTCTTACCAGACTGGTGGAGACAGAAAAACATGAAGTGGTATAAGAAAACAAGCCAGAATCTCAGATGAAATTAAACGCAATGTCAGAGCTAAGTCTGAT
    Name: Uninformative_Sequence_Name_7
    Sequence: GTTATGCGAAATGCACTAAACGCATATAATGGCCAAGCCTAAATTCTTAAGAACTAAGCCCCATCTTAATATAGGCACTATAGGACACATAGACCATGGCAAAACTACATTAACTTCGGCTATAACAAAGTTGTTTGGCAAATACAAATCATATGAGGAAATTGATTATGCTCCAGAGGAAAAAGCTAGGGGCATAACTATAGCTACTGCACATGTTGAATATGAAACTAGCCTAAGACATTATGCCCATATTGATTGCCCTGGCCATGCAGATTATATTAAGAACATGATAACTGGAACAGCTCAAATGGATGGAGCTATATTAGTTTGCTCAGCACTAGATGGACCAATGCCGCAAACCAGAGAGCATATATTACTAGCAAAACAAGTTGGGGTTCCGGCTATTGTTGTGTTTCTTAATAAAATAGATCAAGTTGATGATCCTGAGTTAGTTGAATTAGTAGAATTAGAATTAAGGGAATTGCTGTTGGCATATAATTATCCTGGTAATAACTTGCCCATTATCAGAGGCTCAGCACTTCTAGCACTTCAATCTAATGATGATTGTT
    Name: Uninformative_Sequence_Name_8
    Sequence: TTCCTCGTCGAACTCCTGGGCGGCCAGCAGTTCGCGGACCTCCATCTCGACGAGCTCGAGGAGCTCTTCGTCGTCGACCATGTCGGCCTTGTTCAGCGCGACCAGGATGTAGGGCACGCCGACCTGACGGGCGAGCAGCACGTGCTCCTTCGTCTGGGGCATGGGGCCGTCGGTGGCGGCCACCACCAGGATCGCGCCGTCCATCTGCGCCGCACCCGTGATCATGTTCTTCACATAATCGGCGTGACCGGGGCAATCGACGTGCGCATAGTGGCGGTTCGTCGTCTCATACTCGACGTGAGCCGTCGAGATCGTGATACCGCGAGCCTTCTCTTCCGGCGCCTTGTCGATGTTGGCGAAATCGACCGCAACGCCGCCGCCGGTTTCGGCGAGGACCTTGGTGATGGCCGCCGTCAGCGACGTCTTGCCATGGTCGACGTGACCGATGGTGCCGATGTTGCAGTGCGGCTTGTTCCGCTCGAATTTAGCCTTAGCCATGTT
    Name: Uninformative_Sequence_Name_9
    Sequence: CATAATTCATAAATTATACAATAAATAAAAAGTTTTAATAGGATGCCTGATAAAAGGGTTATTTGATAGAATAAATTATGTATTTTATACTCTTATTATCATTACATTAACAGAATTAAACTGTCATTTAACATTAAAAATTATGTTCTTCATAAACTATAATATAATTTAAAAGATAAGCTAAGAAAGCTAATGGGTTCATACCTCATTTATGGAAACTAAAATCCTCTTTTTAATTAAAAATAATTCATCTCACATAATTTTTTTCACATTTTTAGTTATGGGAATCATGATATCAATCTGTTCTAACAATTGGCTGGGTTGTTGGATAGGAATTGAAGTTAATATGGTTTCCTTTCTACCTTTCATAATTAATAAGATAAGAATTTATTCATCTGAATCAATAATTAAGTATTTCATTATTCAGAGAATGGGTTCAGGATTACTATTTTTTTGTATTATTATTATGGGTTTACATACTCAATTAAAATATATTTTATTAATTAGACTAATAATTAAGATTGGTGCACCTCCATTTCATTTTTGATATGTTTCAGTAGTGGAAGGGTTATCTTGAATAGTAATATTTTTAATAATAACTATTCAAAAAGTGATCCCTATAGTAATGTTAAGTTATCTTAATTCAAGATTAATTGTTTTTGTTGTTATATGTTGTATTTGAGGATCTATTGGAGGGATATGTTATTCTTCTATTCGGAAAATTTTAGCATATTCTTCTATTAATAATTTAGGCTGAATTTTTACTGGAATTATATTGATTAATTACATATGATTAATATATTTTTTAATTTATTCAATATCTTTATATTTAATTTGTTGATTGTTTATAAAAAATAATATTAATTATTTAAATCAATTTTTTGTTCTATTTAGTGATAAAATTGATTTAATTATAATAATGATTATTTTTTTATCAATAGGTGGATTACCACCATTTCTAGGGTTTGTGCCTAAATTATTAATAATTTATTCTCTTGTGATTAATAAAATATATTTTATATGTTTTATTTTAATAATATCGGCTTTAATAGTTTTATTTTATTATTTACGAGTTGTTATAACTGGGTTATTATTGAATATTATCACCTTAAAATTGTCAATTATTAAATCAAAATTCATTTTTTTTATTTCAAATTTTTTATTGATAGGATTAATAATTATTATTTTATTAAATATTTCTAATTAACTAGTTGGAAGGTTTTAAGTTAATTAAACTATTATCCTTCAAAGTTAAAAATATATTTTTATAAACCTTAGATATATTTCAATCACCTTTAAATTTGCAATTTAATATTATAATTAAATATAAGATTAAAGATTATATCGAGATTATGAATATTGAGAGAATATCATTCATGAATAAATTTACAGTTTATTACCTATTTCAGCCATCTCAATTTATATCAATTTTATTTTACTGTAAATTTATGAATAAATGATTTTTTTCAACTAATCATAAAGACATTGGTACATTATATTTTGTCTTTGGAATTTGATCGGGAATAATTGGTACTACTTTAAGAGTTTTAATTCGAATTGAACTAGGTATACCAGGTTCATTTATTGGTGATGATCAAATTTATAATGTGATTGTTACAGCTCATGCTTTTATTATAATTTTTTTTATAGTTATGCCAATTATAATTGGTGGTTTTGGTAATTGACTTGTTCCCTTAATAATTGGGGCTCCTGATATAGCTTTCCCTCGTATAAATAATATAAGTTTTTGATTATTGCCTCCTTCATTAACCTTACTTTTAGTAAGCAGAATAGTTGATAGAGGAGCGGGCACAGGATGAACAGTTTATCCACCTTTATCAAGAAGAGTTGCTCATTCAGGTTCCTGTGTAGATTTAACTATTTTTTCTCTTCATCTGGCAGGTGCAAGATCCATTTTGGGTGCAGTTAATTTTATTAGAACAATTTTTAATATACGTTCTATTGGTATATATATAGATCGAATACCATTATTTGTATGAGCTGTTCTAATTACTGCCTTTTTGCTATTAATTTCTCTTCCTGTTCTGGCAGGTGCAATTACTATACTATTAACTGATCGAAATTTAAATACTTCTTTCTTTGATCCTTCAGGAGGTGGTGATCCTATTTTATATCAGCATTTATTTTGATTTTTTGGGCATCCAGAAGTTTACATTTTAATTTTACCTGGATTTGGTTTGATTTCTCATATTATTACTCAAGAAAGTGGTAAACTTGAATCATTTGGATCATTAGGAATAATTTATGCTATATTATCAATTGGAATTTTAGGATTTGTAGTATGAGCTCATCATATATTTACAGTTGGAATAGATGTAGATACTCGAGCTTATTTCACATCAGCTACTATAATTATTGCTGTACCAACTGGTATTAAAATTTTTAGTTGATTAGCAACCTTAAGAGGTTCAAAAATCAAAATAAATGCATCAATTATGTGAGCTTTAGGATTTGTATTTTTATTTACTATTGGGGGATTAACAGGTATTATTTTAGCTAATTCATCAATTGACATCATTTTACATGATACATATTATGTAGTAGCACATTTTCATTATGTTTTATCAATAGGAGCAGTTTTTGCAATTTTGGGTAGATTTATTCATTGATATCCATTATTTACTGGTTTGTCTCTTAATTCCAATTGATTAAAAATTCATTTTATTATTATATTTATCGGCGTAAATTTAACTTTTTTCCCTCAACATTTTCTAGGATTAAGAGGAATACCTCGACGATATTCAGATTATCCCGATGCATTTATATCATGAAATATTTTATCTTCGGTAGGAAGAACAATTTCTTTAATTGGTATTTTACTATTATTATTTATTGTTTGAGAAAGTTTTATTTCATGTCGTTTTATTTTATATTCAAATAATATAAATTCTTCAATTGAATGATTTCAGAAATTTCCTCCGTCCGAACATTCTTATAATGAAATTCCATTATTAGTAATGTTCTAGTATGGCAGAAAAAGTGCAATGAATTTAAGATTCATATATAAAATTAATTTTTATTAGAAATTTCTAATTGATCTTATATTTATATACAAGATGCTATATCTCCATTAATAGAACAACTTATATTTTTTCATGATCATACATTAGTTATTTTGATTGTAATCACTTCAATTGTTGCTTATATAATAACAACATTATTAATTAATAATTTTGTTAATCATTTGTTATTAGAAGGACAGTTAATTGAATTTTTTTGAACATTATTACCAGCATTAACTTTAATTTTTATTGCACTTCCATCATTACGATTATTATATTTATTAGATGAAATTAATAAACCATTGATTACTTTAAAAATTATTGGTCATCAATGATATTGATCTTATGAATATTCGGATTTTAATAATATTGAATTTGATTCATATATAAAATCATATAGAGATTTATTAATAGAAGAATTTCGTTTATTAGATGTAGATAATCGGACAATTCTACCATTTAACACTCAAATTCGGTTATTAATTTCTTCTTTTGATGTAATTCATTCATGAGCGTTACCATCAATAAGAATTAAACTAGATGCAGTGCCAGGACGATTAAATCAAACTGGAATATTTATTAATCGTCCTGGATTAGCATATGGTCAATGCTCTGAAATTTGTGGGGCAAATCACAGATTTATGCCAATTGTTATTGAAAGAGTTAATTCAAAAAGTTTTATAAATTGATTACTTAATTATTCATTAAGTGACTGAAAGTAAGTAATGGTCTCTTAAACCAATTTATAGTAATTAACGATTGCTCTTTATGATATAAAAAATTAGTTTAATTAAAAATATAAGTCTGTCAGACTTAGGAAGCTGTTGAGGTATTTTTTTATTCCTCAAATATCACCTATATATTGAGTTAGTTTATTTTTTTATTTTGTTTCTAATTTGTTTATAGTATTTAATTTGTATCCATTTTTAATTGCTATTTGTTGGATAATATTTCTTTCTTTGTTTCTGTCTGTGTTGTTTAGTGGGATTCTAAGCATTTGGCATATCATGGAATTAAATACTGCCATTTATGTTGAGGAGGATGGTTAGAATTATTTG
    Name: Uninformative_Sequence_Name_10
    Sequence: AAGAAAGGAAAGGAGCTGATTTCTAGCATTCAAAAATAAAACTGTAATTACTTTGAGCATGATTTTAGGGGGGGGGGAAATATAATGTAATACAGTATAATACATTACATTGTTGTATTTGTAATTTAACTGTCTTTATTGTTACTGTATCTGGTGGGTATGTCTTGTGGTTAATGCATCTTGTCTCTGCTTGTAAATTGAGCATTAAAATCATGAATAGGGAGTCAGATGGATGACCACACAGACTCTCATTGACCAAGGATTTCTACAGTTATAAGAGTAGATAGGGAATACATTAAATTCAGTAATTTTATCATCATCACCAAACAATGATATAAATCCTGTATATAATAGAATGAATAATTATTAATATTAATTCATGTAATATTTATTTAACGATTATTTTTGATTTTATATCAACCTTATTTTTATCAACAGTATCAATTATTTCATCTATAGTAATATTATATAGAGGAGTTTATATAAATAATGATAAATTTATTAATCGGTTCATTTATATTATTATAATATTTGTGTTATCAATAATGTTATTAATTATTAGACCAAATATAATTAGTATTATTATTGGATGAGATGGATTAGGATTAGTTTCATATTGTTTAGTAATTTATTATCAAAATCTAAATTCTAGAAATGCAGGTATATTAACCGCTTTAATAAATCGGGTTGGGGATGTTATAATTTTAATTACTATTGCATGAATGTTAAATTTTGGTAGTTGAAATTTTTTATCATTTATTAATAAAATAGCAATAGGACCTATAATTATATTAGTAATTATTGCTGGATTTACTAAAAGAGCTCAAATTCCTTTTTCTGCATGGTTACCTGCAGCTATAGCTGCTCCAACACCTGTTTCAGCTCTTGTACATTCATCAACACTAGTCACAGCAGGAGTTTACTTAATTATTCGTTTTAGAAAATTAATTAATAGATTTGATTTAATTGAATTATTTATATTAATATCGGTATTGACTATAATTATATCAGGAATAGCCGCAATTTTTGAATTTGATCTAAAAAAAATCATTGCTTTATCAACATTAAGACAATTAGGAATTATAATGTCAATTCTTATATTTGGTTTTCCAATACTTTCATATTTTCATTTAATTATTCATGCTATATTTAAGGCTTCTTTATTTTTATGTGCAGGAATTTTAATTCATAAACTAAATAATAATCAAGATATTCGAATAATAGGATGTATAAGACATAATATACCTTTAACTACAACATTAATAAATATTGCAAATTTATCATTATGTGGAATTCCTTTTATGAGAGGTTTTTATTCAAAAGATTTAATTATAGATATAATATGTTCAAGAAATATTAATGTATTTATTTTGATTTTAATATATTTAGGAATTGGATTAACTTCATTTTATTCAGCCCGATTAACTTTTTTTTCAATAAATATAAATTTTAATTATTCTAATTTTAGATCATTAAATGAACATATAGATAAAATATTAAAAAGAATTTTAATTTTATCTATATATTCAATTTTATCAGGATCAATAATTAGTTGATTATTATTTAATAATATCCCAATTATTATCATTCCATTAGAAGGAAAATTAATAACAATAATTTTATCATTATTAGGAATAGTAATTGGATATGAATTATCAATATTATCAACCATAAAAAAAATATTATTAATAGAATTTATTGGTTCTATATGATTTATGAAACAATTATCTACCTTTCATAACCAATTTTTTGTTCTAAATAATTCATTAATGTTTCAAAAATCAGTTGATCTTGGTTGAGGAGAATTTTATGGTCCTCAAGGATTAGTTCAATTCATAAAATATATTATAAAATATAATATCAAAATAAGAAAAAATAATTTCAAAATTCAAATAATAACATTTATTTTATGATCTATAATTATATTTTATTTAAATAGCTTAAAAAGAGCATAATATTGAAGATATTAAGGTAATAAATTATTTTTAAATACTTACAAGATCAATATCTCTCTTTTCAATGAAAATGAAATGTTTTATAAACTTTATAAATAAGGAAAAAACGGAATTAAACCGCTTTTAAAGCTAGTTAGCAGCTGCTTTTTATAGTTTCCTTTTAATTGGAAGTTAAATTCAATCTCTTTATTAACATTAAAGTGCCTTAGTTTCGGCCTCAATTAATTTAATAATGGAATTATAAATATTCTATATTTAGGCCGAAACTAAGTGTATAAATTACTTCATTATTTCTTTAATAAGAATGGTTGATAACAACACCTATTAATTGCAACTTAATAATTATATTTAAATACATACCTAACTGTTTAAACTCAATTTAATATATTGTTATTTCATTCATGATATAATCCAAAAACCAAGAATATTAAAAATAATATTATAATGACATGAAATTAATAAAATCTATGAAAATAATCCTTATAGAACATTCATTGACAGAGGTATACAAATATATTTAAAGTGAAAACTATCGTGGATTATCAATTAAAAAACAGGTTCCTCTGATTAGATTAATTACCGCCAAATTCTTTGAATTTTAAGATCATAACTAGTAATAATCTGATTTTTTAAATTACGCTTTTAATAATAGGGTATCTAATCCTAGTGCAGTAAAAAATTTTTTAGGCATAAAAATAAATTTTTAATAAAATAATATTTTACCAAAATTGAATTTGTTAATTATTAATGAATTTTTATCACCAAATCGTTTATTGAAAAACTTGGAAAAAGTCAAACTAAAGTTTAAATTTGAATCAGGTCATAGATTATACAGAACCAAATGCTTTTATATTTTCTTTTATTTATTTGAAATACAAATTAAAATATTTCACTGATTTGTTATAAATATTTGTGAAATCTTTATAAACCTGTAAAAATGTATAAACTCCTGATTTACTAATTAATATATTATTTACATAAAAATGTAATTTTGTGTTATTAGATTTCAGAAAGCATAGCATTAATAGGTGCAAAAGAAAATATAGCTTCGCTAGAAAAAGAATACGCAAAAGATGATGTATATTACCTGAAAGTACAAGATATTAGTAACAAATACAAATACATTCGTAGAACACGCCCTGATGGGAACTGCTTCTTTCGATCTTTTGGCTATGCATATTTGGAAAGTTTACTTGATAATACTGAAGAATTTAACAGGTAAAGGTGTAATAACATAGAAATAATAAACAACTTAGTAGACATAAAATTGAAATTATAAATAATTTTTCAGTTCTCACTTATCACTTTATCTCTTTGTGTTGCAACAGTTGTTTATTTAAAAATTCTGTGCTTATTTTGCTTTCAGTTTCCGTGATCTAGCGAAGAAAAGTAAGGAAAAACTTATAGCTCTTGG


Another useful method is to save all of the SeqIO objects into a list. This allows you to refer back to them later (in the for loop you can only go through each object once unless you to the whole loop again)


```python
sequence_record = list(SeqIO.parse("/UCHC/LABS/Simon/Python_Test/mystery.fasta", "fasta"))
print(sequence_record[5].id)
print(sequence_record[5].seq)
```

    Uninformative_Sequence_Name_6
    TCTGTGGCATCTGCCCATCTGTTGCAGCAACAACTAATATAGCACCATCCATCTGAGATGTCCCTGAGATCATGTTTTTAATATAATCTGCATGCCCAGGGCAATCTGTATGAGCATAATGTCTCTTCTTTGTTTTATATTCTATATGTGTAGCATTAATTGTTATGCCACGTGACTTTTCTTCTGGTGCTCTATCAATCTCACTGTATGGAATATACTTGGTCTCGTTTTCTTTTGACATCAGCTTCGTAATTGCTGCAGTAAGAGTAGTTTTTCCATGATCAACATGACCTATGGTTCCCACATTACAATGGTTTCTACTAATTTTCGTACAGTAACACCTTATCCTATCCGTGCTACGTATAAAACTGTAATTACTAGTATTTAAAACTGTTCTTGAACTTTTCACTATATTTAAATACACGACTGTATGCCTCACACATGACAAAACTATGCTGTTCATTTTCTTCAAATTCACTTCAATCCTTTATTTTTCTTGATTGGTCCCACGCCTATATAAAGACGAGAAAAGGAAGACAGGAATATTAGTGTGCTGTAATGAGACTAAGTATGTTGATATTAAAATGGAAGACGTTCTTACCAGACTGGTGGAGACAGAAAAACATGAAGTGGTATAAGAAAACAAGCCAGAATCTCAGATGAAATTAAACGCAATGTCAGAGCTAAGTCTGAT


### Automated Blast
I don't remember where these sequences came from so we are going to blast them and find out
NCBIWWW offers the qblast() function which lets you blast to the ncbi servers with any blast command.
NCBIXML lets you read in the blast object 


You can go through each of your sequences this way but it will take too long for right now. Usually you will submit something like this as a job.


```python
from Bio.Blast import NCBIWWW
from Bio.Blast import NCBIXML
```


```python
for SeqIOobject in sequence_record:
    handle = NCBIWWW.qblast("blastn", "nt", SeqIOobject.seq)
    blast_record = NCBIXML.read(handle)
    
    for alignment in blast_record.alignments:
        print(alignment.title)
        # do whatever
        
```

    gi|1406948775|gb|MG737759.1| Tryella crassa isolate TRYCRA mitochondrion, partial genome
    gi|1406948439|gb|MG737735.1| Aleeta curvicosta isolate ALECUR mitochondrion, partial genome
    gi|1406948523|gb|MG737741.1| Magicicada septendecim isolate MAGSEP mitochondrion, partial genome
    gi|1406948509|gb|MG737740.1| Magicicada neotredecim isolate MAGNEO mitochondrion, partial genome
    gi|1406948747|gb|MG737757.1| Magicicada tredecula isolate MAGTDC mitochondrion, partial genome
    gi|1406948537|gb|MG737742.1| Magicicada septendecula isolate MAGSDC mitochondrion, partial genome
    gi|676260431|gb|KM000130.1| Magicicada tredecim mitochondrion, partial genome
    gi|1406948733|gb|MG737756.1| Magicicada tredecim isolate MAGTRE13 mitochondrion, partial genome
    gi|1406948719|gb|MG737755.1| Magicicada tredecim isolate MAGTRE12 mitochondrion, partial genome
    gi|1406948705|gb|MG737754.1| Magicicada tredecim isolate MAGTRE11 mitochondrion, partial genome
    gi|1406948691|gb|MG737753.1| Magicicada tredecim isolate MAGTRE10 mitochondrion, partial genome
    gi|1406948649|gb|MG737750.1| Magicicada tredecim isolate MAGTRE07 mitochondrion, partial genome
    gi|1406948635|gb|MG737749.1| Magicicada tredecim isolate MAGTRE06 mitochondrion, partial genome
    gi|1406948607|gb|MG737747.1| Magicicada tredecim isolate MAGTRE04 mitochondrion, partial genome
    gi|1406948593|gb|MG737746.1| Magicicada tredecim isolate MAGTRE03 mitochondrion, partial genome
    gi|1406948579|gb|MG737745.1| Magicicada tredecim isolate MAGTRE02 mitochondrion, partial genome
    gi|1406948565|gb|MG737744.1| Magicicada tredecim isolate MAGTRE01 mitochondrion, partial genome
    gi|1406948677|gb|MG737752.1| Magicicada tredecim isolate MAGTRE09 mitochondrion, partial genome
    gi|1406948663|gb|MG737751.1| Magicicada tredecim isolate MAGTRE08 mitochondrion, partial genome
    gi|1406948621|gb|MG737748.1| Magicicada tredecim isolate MAGTRE05 mitochondrion, partial genome
    gi|1406948271|gb|MG737723.1| Kosemia yezoensis isolate KOSYEZ mitochondrion, partial genome
    gi|1406948453|gb|MG737736.1| Amphipsalta zelandica isolate AMPZEL mitochondrion, partial genome
    gi|1406948467|gb|MG737737.1| Kikihia muta isolate KIKMUT mitochondrion, partial genome
    gi|1406948481|gb|MG737738.1| Kikihia ochrina isolate KIKOCH mitochondrion, partial genome
    gi|1406948845|gb|MG737764.1| Cicadettana calliope isolate CICCAL mitochondrion, partial genome
    gi|1406948355|gb|MG737729.1| Muda kuroiwae isolate MUDKUR mitochondrion, partial genome
    gi|1406948411|gb|MG737733.1| Yezoterpnosia vacua isolate TERVAC mitochondrion, partial genome
    gi|1406948803|gb|MG737761.1| Okanagana canadensis isolate OKACAN mitochondrion, partial genome
    gi|1406948831|gb|MG737763.1| Okanagana rimosa isolate DB119 mitochondrion, partial genome
    gi|1406948215|gb|MG737719.1| Euterpnosia chibensis isolate EUTCHI mitochondrion, partial genome
    gi|1406948817|gb|MG737762.1| Okanagana oregona isolate DB114 mitochondrion, partial genome
    gi|1406948789|gb|MG737760.1| Okanagana bella isolate DB118 mitochondrion, partial genome
    gi|1406948397|gb|MG737732.1| Yezoterpnosia nigricosta isolate TERNIG mitochondrion, partial genome
    gi|1406948341|gb|MG737728.1| Mogannia minuta isolate MONMIN mitochondrion, partial genome
    gi|1406948187|gb|MG737717.1| Cryptotympana atrata isolate CRYATR mitochondrion, partial genome
    gi|1406948551|gb|MG737743.1| Magicicada tredecassini isolate MAGTCS mitochondrion, partial genome
    gi|1406948495|gb|MG737739.1| Magicicada cassini isolate MAGCAS mitochondrion, partial genome
    gi|1406949349|gb|MG737800.1| Tettigades undata isolate PL811x9 mitochondrion, partial genome
    gi|1406949335|gb|MG737799.1| Tettigades undata isolate PL811x8 mitochondrion, partial genome
    gi|1406949321|gb|MG737798.1| Tettigades undata isolate PL811x6 mitochondrion, partial genome
    gi|1406949293|gb|MG737796.1| Tettigades undata isolate PL809x1 mitochondrion, partial genome
    gi|1318752673|gb|MG674196.1| Tettigades undata isolate TETLON mitochondrion, partial genome
    gi|1406949307|gb|MG737797.1| Tettigades undata isolate PL811x5 mitochondrion, partial genome
    gi|1406949545|gb|MG737814.1| Tettigotoma maculata isolate PL799 mitochondrion, partial genome
    gi|1406949265|gb|MG737794.1| Tettigades undata isolate JM032 mitochondrion, partial genome
    gi|1406949251|gb|MG737793.1| Tettigades undata isolate JM028 mitochondrion, partial genome
    gi|1406949377|gb|MG737802.1| Babras sonorivox isolate PL622 mitochondrion, partial genome
    gi|676260445|gb|KM000131.1| Diceroprocta semicincta mitochondrion, partial genome
    gi|1406948257|gb|MG737722.1| Hyalessa maculaticollis isolate HYAMAC mitochondrion, partial genome
    gi|426262354|gb|JQ910987.1| Oncotympana maculaticollis mitochondrion, partial genome



    ---------------------------------------------------------------------------

    KeyboardInterrupt                         Traceback (most recent call last)

    <ipython-input-9-405765f8d7bd> in <module>
          3 
          4 for SeqIOobject in sequence_record:
    ----> 5     handle = NCBIWWW.qblast("blastn", "nt", SeqIOobject.seq)
          6     blast_record = NCBIXML.read(handle)
          7 


    /UCHC/LABS/Simon/venv/py-3.6.3/lib/python3.6/site-packages/Bio/Blast/NCBIWWW.py in qblast(program, database, sequence, url_base, auto_format, composition_based_statistics, db_genetic_code, endpoints, entrez_query, expect, filter, gapcosts, genetic_code, hitlist_size, i_thresh, layout, lcase_mask, matrix_name, nucl_penalty, nucl_reward, other_advanced, perc_ident, phi_pattern, query_file, query_believe_defline, query_from, query_to, searchsp_eff, service, threshold, ungapped_alignment, word_size, alignments, alignment_view, descriptions, entrez_links_new_window, expect_low, expect_high, format_entrez_query, format_object, format_type, ncbi_gi, results_file, show_overview, megablast, template_type, template_length)
        177         wait = qblast._previous + delay - current
        178         if wait > 0:
    --> 179             time.sleep(wait)
        180             qblast._previous = current + wait
        181         else:


    KeyboardInterrupt: 



```python
## Theoretically this works but I haven't tested it
fasta_file = open("/UCHC/LABS/Simon/Python_Test/mystery.fasta", "r").read()
handle = NCBIWWW.qblast("blastn", "nt", fasta_file)
blast_record_2 = NCBIXML.parse(handle)

for blast_object in blast_record_2:
    for alignment in blast_object.alignments:
        print(alignment.title)
    
```

Instead we are all going to blast one of the mystery sequences:


```python
handle = NCBIWWW.qblast("blastn", "nt", sequence_record[0].seq)
blast_record_3 = NCBIXML.read(handle)
for alignment in blast_record_3.alignments:
    for hsp in alignment.hsps:
        if hsp.expect < 0.0001:
            print(alignment.title)
            print("length:", alignment.length)
            print("e-value:", hsp.expect)
            print(hsp.identities)
            #if you want to print the whole alignment, you have to break it into chunks
            print(hsp.query[0:50], "...")
            print(hsp.match[0:50], "...")
            print(hsp.sbjct[0:50], "...")
            print()
        
```

Protein blast search. You could also use blastx for this but I wanted to share the method for translating DNA to proteins.

Protein outputs look ugly because it clumps things together for some reason


```python
def translate(sequence):
    table = 11
    min_pro_len = 100
    prolist = []

    for strand, nuc in [(+1, sequence), (-1, sequence.reverse_complement())]:
        for frame in range(3):
            length = 3 * ((len(sequence)-frame) // 3)
            for pro in nuc[frame:frame+length].translate(table).split("*"):
                prolist.append(pro)
    return(max(prolist, key=len))

#handle = NCBIWWW.qblast("blastp", "nr", translate(sequence_record[0].seq))
#blast_record_4 = NCBIXML.read(handle)
for alignment in blast_record_4.alignments:
    for hsp in alignment.hsps:
        if hsp.expect < 0.0001:
            print(alignment.title)
            print("length:", alignment.length)
            print("e-value:", hsp.expect)
            print("identities:", hsp.identities)
            print(hsp.query[0:50], "...")
            print(hsp.match[0:50], "...")
            print(hsp.sbjct[0:50], "...")
            print()
```

    gi|1406948510|gb|AWV83484.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada neotredecim] >gi|1406948524|gb|AWV83497.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada septendecim]
    length: 323
    e-value: 1.83572e-45
    identities: 78
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFLIYS SL+L CYL   NN+NYLNQ F++F NKI + +++++F+S+G ...
    LVYFLIYSLSLFLTCYLFWTNNMNYLNQIFILFNNKITSLMLMVIFMSMG ...
    
    gi|676260432|gb|AIM19523.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim]
    length: 315
    e-value: 1.07156e-44
    identities: 76
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFLIYS SL+L CYL   NN+NYLNQ F++F NKI + +++++F+S+G ...
    LVYFLIYSLSLFLTCYLFWTNNMNYLNQIFILFNNKITSLMLMVIFMSMG ...
    
    gi|1406948566|gb|AWV83536.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948580|gb|AWV83549.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948594|gb|AWV83562.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948608|gb|AWV83575.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948622|gb|AWV83588.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948636|gb|AWV83601.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948650|gb|AWV83614.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948664|gb|AWV83627.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948678|gb|AWV83640.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948692|gb|AWV83653.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948706|gb|AWV83666.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948720|gb|AWV83679.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim] >gi|1406948734|gb|AWV83692.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecim]
    length: 323
    e-value: 1.33319e-44
    identities: 76
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFLIYS SL+L CYL   NN+NYLNQ F++F NKI + +++++F+S+G ...
    LVYFLIYSLSLFLTCYLFWTNNMNYLNQIFILFNNKITSLMLMVIFMSMG ...
    
    gi|1406948440|gb|AWV83419.1| NADH dehydrogenase subunit 2 (mitochondrion) [Aleeta curvicosta]
    length: 323
    e-value: 4.35946e-44
    identities: 79
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFL+Y FSL+L CYL  +NN+NYLNQ F +F NKI + +++IVF+S+G ...
    LVYFLVYLFSLFLTCYLFKMNNMNYLNQVFTLFNNKISSLMLMIVFMSMG ...
    
    gi|1406948776|gb|AWV83731.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tryella crassa]
    length: 323
    e-value: 7.63144e-44
    identities: 77
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFLIY FSL+L CYL  +NN+NYLNQ F +F NKI +F+++I+F+S+G ...
    LVYFLIYLFSLFLTCYLFKMNNMNYLNQIFTLFNNKIDSFMLMIIFMSMG ...
    
    gi|1406948538|gb|AWV83510.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada septendecula] >gi|1406948748|gb|AWV83705.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecula]
    length: 323
    e-value: 9.57283e-43
    identities: 74
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFLIYS SL+L CYL  +NN+NYLNQ F++F N +K+ +++I+F+S+G ...
    LVYFLIYSLSLFLTCYLFWMNNMNYLNQVFILFNNSVKSLMLMIIFMSMG ...
    
    gi|1406948496|gb|AWV83471.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada cassini] >gi|1406948552|gb|AWV83523.1| NADH dehydrogenase subunit 2 (mitochondrion) [Magicicada tredecassini]
    length: 323
    e-value: 1.07754e-42
    identities: 75
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YFLIYS SL+L CYL  +NN+NYLNQ F++F N IK+ +++I+F+S+G ...
    LVYFLIYSLSLFLTCYLFWMNNMNYLNQIFILFNNSIKSLMLMIIFMSMG ...
    
    gi|1406948468|gb|AWV83445.1| NADH dehydrogenase subunit 2 (mitochondrion) [Kikihia muta]
    length: 324
    e-value: 4.35149e-33
    identities: 64
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    LIY+ IYSF+L  +CY+  + N+NY+NQF ++  N +K+ +++ +F+S+G ...
    LIYYFIYSFTLLAVCYMFKMFNINYINQFIMVSFNFMKSIMMMCIFMSMG ...
    
    gi|1406949532|gb|AWV84433.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades sarcinatrix]
    length: 326
    e-value: 4.19211e-32
    identities: 65
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSLTLSMLCYLFSLNNLNYLNQFFLIYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406949518|gb|AWV84420.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades sarcinatrix]
    length: 326
    e-value: 4.96987e-32
    identities: 65
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSLTLSMLCYLFSLNNLNYLNQFFLIYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406948944|gb|AWV83887.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades distanti]
    length: 326
    e-value: 7.68619e-32
    identities: 62
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CY+   NNLNYLNQFF+++ N     +++++F+S+GG ...
    IYFLIYSLTLFMLCYMFSFNNLNYLNQFFMVYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406949140|gb|AWV84069.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades opaca]
    length: 326
    e-value: 8.1059e-32
    identities: 61
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CYL  +NNLNYLNQFFLI+ N     +V+++F+S+GG ...
    IYFLIYSMTLFMLCYLFSLNNLNYLNQFFLIYSNFYSWLVVMMLFMSMGG ...
    
    gi|1406949252|gb|AWV84173.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata] >gi|1406949280|gb|AWV84199.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata]
    length: 326
    e-value: 1.47014e-31
    identities: 65
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYSF+L ++C L   NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSFTLSMLCCLFSFNNLNYLNQFFLIYNNFYSWLIMMMLFMSMGG ...
    
    gi|1406949266|gb|AWV84186.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata]
    length: 326
    e-value: 1.48584e-31
    identities: 65
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYSF+L ++C L   NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSFTLSMLCCLFSFNNLNYLNQFFLIYNNFYSWLIMMMLFMSMGG ...
    
    gi|1406948930|gb|AWV83874.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades distanti]
    length: 326
    e-value: 2.58211e-31
    identities: 63
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CY+   NNLNYLNQ F+++ N     +++++F+S+GG ...
    IYFLIYSLTLFMLCYMFSFNNLNYLNQLFMVYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406948958|gb|AWV83900.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades distanti]
    length: 326
    e-value: 2.63757e-31
    identities: 63
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CY+   NNLNYLNQ F+++ N     +++++F+S+GG ...
    IYFLIYSLTLFMLCYMFSFNNLNYLNQLFMVYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406948482|gb|AWV83458.1| NADH dehydrogenase subunit 2 (mitochondrion) [Kikihia ochrina]
    length: 324
    e-value: 4.22476e-31
    identities: 61
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    LIY+ +YSF+L  +CY+  + N+NY+NQF ++  N +K+ +++ +F+S+G ...
    LIYYFVYSFTLLAVCYMFKMFNINYINQFIMVSFNFMKSIMMMCIFMSMG ...
    
    gi|1406949154|gb|AWV84082.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades sp. 1 PL2017] >gi|1406949168|gb|AWV84095.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades sp. 1 PL2017]
    length: 326
    e-value: 6.43746e-31
    identities: 62
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSLTLSMLCYLFSLNNLNYLNQFFLIYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406948846|gb|AWV83796.1| NADH dehydrogenase subunit 2 (mitochondrion) [Cicadettana calliope]
    length: 324
    e-value: 8.70048e-31
    identities: 55
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    LIY+ IYSF+L  +CY+  + N+NY+NQF +   N +K+ +V+ +F+S+G ...
    LIYYFIYSFTLMAVCYMFSLFNINYINQFIMTSTNFVKSSMVMCIFMSMG ...
    
    gi|1406948454|gb|AWV83432.1| NADH dehydrogenase subunit 2 (mitochondrion) [Amphipsalta zelandica]
    length: 324
    e-value: 9.17501e-31
    identities: 62
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+Y+ IYSF+L  +CY+  + N+NY+NQF L   + +K+ +++ +F+S+G ...
    LVYYFIYSFTLMAVCYMFSLFNINYINQFILSSFDFMKSIMMMCIFMSMG ...
    
    gi|1406949238|gb|AWV84160.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata]
    length: 326
    e-value: 9.23637e-31
    identities: 64
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYSF+L ++C L   NNLNYLNQ FLI+ N     +++++F+S+GG ...
    IYFLIYSFTLSMLCCLFSFNNLNYLNQLFLIYNNFYSWLIMMMLFMSMGG ...
    
    gi|1406949210|gb|AWV84134.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades ulnaria]
    length: 326
    e-value: 9.53529e-31
    identities: 59
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CYL + NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSLTLFMLCYLFNSNNLNYLNQFFLIYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406949364|gb|AWV84277.1| NADH dehydrogenase subunit 2 (mitochondrion) [Alarcta quadrimacula]
    length: 326
    e-value: 1.33925e-30
    identities: 58
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYF+IYS +L+++CYL +++NLNYLNQ FLI+ N     L++++FLS+GG ...
    IYFIIYSMTLFMMCYLFNLHNLNYLNQVFLIYNNFYDWLLMMVLFLSMGG ...
    
    gi|1406948384|gb|AWV83367.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tanna japonensis]
    length: 323
    e-value: 1.85289e-30
    identities: 59
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    ++Y+ IYS +L ++CY   I  LNYLNQFFLI  N +K   V+++FLS+G ...
    MMYYAIYSLTLMMVCYSFYIFGLNYLNQFFLISMNYMKLIFVMVIFLSMG ...
    
    gi|1318752674|gb|AUJ22103.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata] >gi|1406949294|gb|AWV84212.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata] >gi|1406949308|gb|AWV84225.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata] >gi|1406949322|gb|AWV84238.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata] >gi|1406949336|gb|AWV84251.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata] >gi|1406949350|gb|AWV84264.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades undata]
    length: 326
    e-value: 3.19704e-30
    identities: 64
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++C L   NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSLTLLMLCCLFSFNNLNYLNQFFLIYNNFYSWLIMMVLFMSMGG ...
    
    gi|1406949546|gb|AWV84446.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigotoma maculata]
    length: 325
    e-value: 5.33168e-30
    identities: 60
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    +YFLIYS +L +ICYL ++ N+NYLNQFFLI+ N   + +++++F+S+GG ...
    LYFLIYSMTLCMICYLFNLYNVNYLNQFFLIYNNFYSSLIMMVLFMSMGG ...
    
    gi|1406948412|gb|AWV83393.1| NADH dehydrogenase subunit 2 (mitochondrion) [Yezoterpnosia vacua]
    length: 323
    e-value: 5.89249e-30
    identities: 64
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    +IY+LIYS +L ++CY   I  +NYLNQFFLI  N +K   ++I+FLS+G ...
    IIYYLIYSLTLMMVCYSFYIFGMNYLNQFFLISMNYMKLIFMMIIFLSMG ...
    
    gi|1406948902|gb|AWV83848.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades chilensis]
    length: 326
    e-value: 6.23626e-30
    identities: 56
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFFLI+ +     +++++F+S+GG ...
    IYFLIYSSTLSMLCYLFSLNNLNYLNQFFLIYNDLYSWLIMMVLFMSMGG ...
    
    gi|1318752632|gb|AUJ22064.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades chilensis]
    length: 326
    e-value: 6.36992e-30
    identities: 56
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFFLI+ +     +++++F+S+GG ...
    IYFLIYSSTLSMLCYLFSLNNLNYLNQFFLIYNDLYSWLIMMVLFMSMGG ...
    
    gi|1406948398|gb|AWV83380.1| NADH dehydrogenase subunit 2 (mitochondrion) [Yezoterpnosia nigricosta]
    length: 323
    e-value: 9.80475e-30
    identities: 63
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    +IY+LIYS +L ++CY   I  +NYLNQFFL   N +K   ++I+FLS+G ...
    IIYYLIYSLTLMMVCYSFYIFGMNYLNQFFLTSMNFMKLIFMMIIFLSMG ...
    
    gi|1406949126|gb|AWV84056.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades opaca]
    length: 326
    e-value: 1.11723e-29
    identities: 62
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFF I+ N     +++++F+S+GG ...
    IYFLIYSMTLCMLCYLFSLNNLNYLNQFFFIYNNFYSWLILMMLFMSMGG ...
    
    gi|1318752646|gb|AUJ22077.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades chilensis] >gi|1406948916|gb|AWV83861.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades chilensis]
    length: 326
    e-value: 1.12914e-29
    identities: 56
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L ++CYL  +NNLNYLNQFFLI+ +     +++++F+S+GG ...
    IYFLIYSLTLSMLCYLFSLNNLNYLNQFFLIYNDLYSWLIMMVLFMSMGG ...
    
    gi|1406949392|gb|AWV84303.1| NADH dehydrogenase subunit 2 (mitochondrion) [Calliopsida cinnabarina]
    length: 326
    e-value: 1.3237e-29
    identities: 53
    YFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGGL ...
    YFLIYS +L+++CYL ++ NLNYLNQFF +      + +++++F+S+GG+ ...
    YFLIYSMTLFMVCYLFNLYNLNYLNQFFTLHNGFYSSLMMMVLFMSMGGM ...
    
    gi|1406948272|gb|AWV83263.1| NADH dehydrogenase subunit 2 (mitochondrion) [Kosemia yezoensis]
    length: 324
    e-value: 1.7525e-29
    identities: 59
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+Y+ IYSF+L  +CY+ ++ N+NY+NQ  +   + +K+ +++ VF+S+G ...
    LVYYFIYSFTLMAMCYMFNLFNINYINQLIMTSHDFLKSLMMMSVFMSLG ...
    
    gi|1406949490|gb|AWV84394.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades dumfriesi]
    length: 325
    e-value: 6.16622e-29
    identities: 59
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    LIY+L+YS +L+++CYL  + NLNYLNQFFLI+ N     L +++F+S+G ...
    LIYYLVYSLTLFMLCYLFSLYNLNYLNQFFLIYNNFYSWLLTMVLFMSMG ...
    
    gi|1406949420|gb|AWV84329.1| NADH dehydrogenase subunit 2 (mitochondrion) [Chonosia atrodorsalis]
    length: 326
    e-value: 6.62171e-29
    identities: 56
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CY+    N+NYLNQFF+I+       +++ +F+S+GG ...
    IYFLIYSLTLFMVCYMFSSYNVNYLNQFFMIYNGFYSWLIMMTLFMSMGG ...
    
    gi|1239292293|gb|ASW26888.1| NADH dehydrogenase subunit 2 (mitochondrion) [Platypleura kaempferi]
    length: 323
    e-value: 1.09842e-28
    identities: 57
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YF IYSF+L+++CY++++  +NYLNQ F+   N   + +++ +F+S+G ...
    LLYFFIYSFTLFMVCYMMNMYEINYLNQMFIFSKNYFLSTMMMSIFMSMG ...
    
    gi|1406948230|gb|AWV83224.1| NADH dehydrogenase subunit 2 (mitochondrion) [Graptopsaltria bimaculata]
    length: 323
    e-value: 1.27392e-28
    identities: 64
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    +IY++IYS +L ++CY   I  +NYLNQFFL+  +  K   ++I+FLS+G ...
    MIYYIIYSLTLMMVCYSFYIFGMNYLNQFFLLSMSFNKLLFMMIIFLSMG ...
    
    gi|1406948216|gb|AWV83211.1| NADH dehydrogenase subunit 2 (mitochondrion) [Euterpnosia chibensis]
    length: 323
    e-value: 1.28747e-28
    identities: 60
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    +IY+LIYS +L ++CY   I  +NYLNQFFL+  + +K   ++I+FLS+G ...
    IIYYLIYSLTLMMVCYSFYIFGMNYLNQFFLVSMDYMKLIFMMIIFLSMG ...
    
    gi|1406948790|gb|AWV83744.1| NADH dehydrogenase subunit 2 (mitochondrion) [Okanagana bella]
    length: 326
    e-value: 1.41865e-28
    identities: 61
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    +YF IY+ +L+++C+L ++ NLNYLNQFFL++ N     +++I+F+S+GG ...
    LYFFIYALTLFMVCFLFNLYNLNYLNQFFLLYNNFHDWLIMMILFMSMGG ...
    
    gi|1406949182|gb|AWV84108.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades sp. 2 PL2017]
    length: 326
    e-value: 2.12052e-28
    identities: 65
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    IYFLIYS +L+++CYL   NNLNYLNQFFLI+ N     +++++F+S+GG ...
    IYFLIYSVTLFMLCYLFSFNNLNYLNQFFLIYNNFYSWLVMMVLFMSMGG ...
    
    gi|1406948370|gb|AWV83354.1| NADH dehydrogenase subunit 2 (mitochondrion) [Platypleura kaempferi] >gi|1406949574|gb|AWV84472.1| NADH dehydrogenase subunit 2 (mitochondrion) [Platypleura kaempferi]
    length: 323
    e-value: 2.13989e-28
    identities: 57
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L+YF IYSF+L+++CY++++  +NYLNQ F+   N   + +++ +F+S+G ...
    LLYFFIYSFTLFMVCYMMNMYEINYLNQMFIFSKNYFLSTIMMSIFMSMG ...
    
    gi|1406948832|gb|AWV83783.1| NADH dehydrogenase subunit 2 (mitochondrion) [Okanagana rimosa]
    length: 326
    e-value: 2.85126e-28
    identities: 61
    YFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGGL ...
    YF IY+ +L+++C+L  + NLNYLNQFFL++ N     + +I+F+S+GGL ...
    YFFIYTLTLFMVCFLFSLYNLNYLNQFFLLYNNFYDWLITMILFMSMGGL ...
    
    gi|1406949476|gb|AWV84381.1| NADH dehydrogenase subunit 2 (mitochondrion) [Platypedia putnami]
    length: 326
    e-value: 4.54e-28
    identities: 55
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    +YF+IYS +++++CY   +NNLNYLNQFFL++       +++I+F+S+GG ...
    LYFVIYSLNIFMLCYFFYLNNLNYLNQFFLVYKKFYNWAIMMILFMSMGG ...
    
    gi|702076320|gb|AIW06145.1| NADH dehydrogenase subunit 2 (mitochondrion) [Gaeana maculata]
    length: 323
    e-value: 5.78439e-28
    identities: 60
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    +IY+LIYS +L ++CYL  I  LNYLNQ FLI  + +K  L+ I+FLS+G ...
    MIYWLIYSLTLMMVCYLFYIFGLNYLNQLFLISISFMKLILLAIIFLSMG ...
    
    gi|1406948342|gb|AWV83328.1| NADH dehydrogenase subunit 2 (mitochondrion) [Mogannia minuta]
    length: 323
    e-value: 1.26475e-27
    identities: 63
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    ++YF+IYS SL LICY+++ NN+NYLNQFF    N  K  + ++V LS+G ...
    MLYFIIYSMSLMLICYMMNDNNINYLNQFFYSM-NLYKMIISMMVLLSMG ...
    
    gi|1406949504|gb|AWV84407.1| NADH dehydrogenase subunit 2 (mitochondrion) [Tettigades major]
    length: 326
    e-value: 2.70585e-27
    identities: 55
    IYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGG ...
    +YFLIYS +L+++CYL  +NNLNYLNQFFL++ N     +++++F+S+GG ...
    VYFLIYSSTLFMLCYLFYLNNLNYLNQFFLVYSNFYIWLILMMLFMSMGG ...
    
    gi|1406948356|gb|AWV83341.1| NADH dehydrogenase subunit 2 (mitochondrion) [Muda kuroiwae]
    length: 325
    e-value: 3.38966e-27
    identities: 60
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    ++YF+IYS +L+++CY  ++ N+NY+NQFF+   N +K+  +++ FLSIG ...
    MMYFIIYSINLFMMCYFFNLMNINYMNQFFMCSDNMVKSLFLMVNFLSIG ...
    
    gi|1406948804|gb|AWV83757.1| NADH dehydrogenase subunit 2 (mitochondrion) [Okanagana canadensis]
    length: 325
    e-value: 3.42562e-27
    identities: 59
    YFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIGGL ...
    YF IY+ +L+++C+L ++ NLNYLNQ FL++ N     +++I+F+S+GGL ...
    YFFIYALTLFMVCFLFNLYNLNYLNQLFLLYNNFHDWLILMILFMSMGGL ...
    
    gi|676260446|gb|AIM19536.1| NADH dehydrogenase subunit 2 (mitochondrion) [Diceroprocta semicincta]
    length: 326
    e-value: 5.37251e-27
    identities: 55
    LIYFLIYSFSLYLICYLLDINNLNYLNQFFLIFGNKIKTFLVIIVFLSIG ...
    L YF IYS +L+++CY  +  N+NYLNQFF+     + + L++++F+S+G ...
    LFYFSIYSLNLFMVCYTFNSLNVNYLNQFFVFSKIYMNSLLMMVLFMSMG ...
    


### Blast parsing and Entrez

To make these results useful, we want the code to parse through all the hits for every sequence and choose a consensus identity. You can do this using ncbi's taxonomy database. 

Every ncbi database has a different type of accession number. The taxonomy database uses taxid's but it won't take gi numbers or gb numbers. 

We can query the gb database using the gb numbers and then extract any info we want about our sequences


```python
from Bio import Entrez
import re

Entrez.email = "jason.vailionis@uconn.edu"
tax_list = []

for alignment in blast_record_3.alignments:
    for hsp in alignment.hsps:
        if hsp.expect < 0.0001:
            gb_number = re.findall("(gb\|.*)\|", alignment.title)[0]
            handle = Entrez.efetch(db="Nucleotide", id=gb_number, retmode="xml")
            record = Entrez.read(handle)
            handle.close()
            #The parse method isn't working correctly for me so I brute forced it
            #it's supposed to look like this:
            #print(record['GBSeq_taxonomy'])
            lineage = re.findall("'GBSeq_taxonomy': '([A-Za-z; /-]*)'", str(record))
            organism = re.findall("'GBSeq_organism': '([A-Za-z /-]*)'", str(record))
            tax_list.append(lineage[0] + "; " + organism[0])
        
        #print(tax_list)
        ordered_taxonomy = tax_list[0]
        split_tax_list = [set(a.split("; ")) for a in tax_list]
        
        shared_list = list(split_tax_list[0].intersection(*split_tax_list[1:]))
        for taxa in shared_list:
            shared_index.append(ordered_taxonomy.index(taxa))
        print(ordered_taxonomy[max[shared_index]]
            
            
```

    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Taphurini', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Taphurini', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Taphurini', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Taphurini', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Taphurini', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]
    [{'Tryella crassa', 'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Tryella', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Aleeta curvicosta', 'Aleeta', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada neotredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada septendecula', 'Hexapoda', 'Arthropoda', 'decula group', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Taphurini', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}, {'Neoptera', 'Metazoa', 'Auchenorrhyncha', 'Paraneoptera', 'Cicadettinae', 'Cicadidae', 'Hemiptera', 'Insecta', 'Lamotialnini', 'Magicicada tredecim', 'decim group', 'Hexapoda', 'Arthropoda', 'Cicadoidea', 'Eukaryota', 'Pterygota', 'Ecdysozoa', 'Magicicada'}]



    ---------------------------------------------------------------------------

    KeyboardInterrupt                         Traceback (most recent call last)

    <ipython-input-14-3bc81b1e02ac> in <module>
          9         if hsp.expect < 0.0001:
         10             gb_number = re.findall("(gb\|.*)\|", alignment.title)[0]
    ---> 11             handle = Entrez.efetch(db="Nucleotide", id=gb_number, retmode="xml")
         12             record = Entrez.read(handle)
         13             handle.close()


    /UCHC/LABS/Simon/venv/py-3.6.3/lib/python3.6/site-packages/Bio/Entrez/__init__.py in efetch(db, **keywords)
        182             # more than about 200 IDs
        183             post = True
    --> 184     return _open(cgi, variables, post=post)
        185 
        186 


    /UCHC/LABS/Simon/venv/py-3.6.3/lib/python3.6/site-packages/Bio/Entrez/__init__.py in _open(cgi, params, post, ecitmatch)
        527     wait = _open.previous + delay - current
        528     if wait > 0:
    --> 529         time.sleep(wait)
        530         _open.previous = current + wait
        531     else:


    KeyboardInterrupt: 





![image.png](attachment:image.png)

http://biopython.org/DIST/docs/tutorial/images/BlastRecord.png




output to a file


next time: bigger datasets, alignments, entrez, ...
to resolve some of them, we can use this method I made which will use ENTREZ to get the lineage and do the lca thing that I made...



